### Investigate null values in item_type_code_name

In [41]:
%run '../lib/libraries.ipynb'

dython              0.6.1
tinydb              4.2.0


You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.
You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.


In [42]:
filterDf = load_df('filterDf')

###### Create a dataframe of null observations

In [43]:
nullDf = filterDf.filter(F.col('item_type_code_name').isNull())

nullDf.count()

3264

3,264 observations
- close to the 3298 nulls in response correctness

###### Display distinct values

In [44]:
distinct_val(nullDf)

0
assessment_instance_attempt_id (aa)            NaN
assessment_item_response_id (ia)               NaN
assigned_item_status (ia)                 assigned
assignment_attempt_number (aa)                   1
assignment_max_attempts (a)                      1
is_manual_scoring_required (aa)              False
item_attempt_end_datetime_utc (ia)             NaT
item_attempt_start_datetime_utc (ia)           NaT
item_is_offline_scored (i)                   False
item_type_code_name (i)                       None
raw_score (ia)                                 NaN
response_correctness (ia)                     None
scoring_type_code (i)                 [unassigned]

- Not attempted
  - assessment_instance_attempt_id = Null (NaN)
  - raw_score = null (Nan)

- item_type_code_name may be set when attempted

##### Conclusion
- null values are correct


In [45]:
finish_todo('Investigate null values in item_type_code_name')